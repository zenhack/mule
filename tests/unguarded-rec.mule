# This should fail, as the type is meaningless. It does, but we should
# report better errors.
4 : rec a. a
