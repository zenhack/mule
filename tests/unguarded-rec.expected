
Desugared:

	(: (Integer 4) (rec a a))
Type error: Unguarded recursive type `rec a. a`. Recursively-bound type variables must be "guarded" by a union, record, or function.
